assignment/impl/src/java/org/sakaiproject/assignment/impl/AssignmentServiceImpl.java

```diff
@@ -451,9 +452,18 @@ public void handleAccess(HttpServletRequest req, HttpServletResponse res, Refere
                     // fail the request, user not logged in yet.
                 } else {
                     if (REF_TYPE_SUBMISSIONS.equals(ref.getSubType())) {
+                        String assignmentName = "bulk_download";
+                        try {
+                            assignmentName = getAssignment(ref.getReference()).getTitle();
+                        } catch (Exception ignore) {
+                            log.warn("Could not find assignment for ref = {}", ref.getReference());
+                        }
+
+                        SimpleDateFormat formatter = new SimpleDateFormat("yyyy-MM-dd_HH-mm");
```

Comment:
I think it'd be easier and more standard to just combine the time. like
"yyyyMMdd'T'HHmmss"

20171122T113147

https://en.wikipedia.org/wiki/ISO_8601

Reply:
Not very readable IMO.

Reply:
It really doesn't matter, as long as the last part is unique for each save/zip. The OS will handle the creation date too.
We have used this change since 10 after we had an instructor request to include the assignment title and a timestamp.

Reply:
Right, I don't think the readability matters either as long as it's unique. I'd be okay with just getting rid of the colons.

"yyyy-MM-dd'T'HHmmss"

Reply:
Sounds good but lets use the newer java 8 time methods, DateTimeFormatter.